rabbitmq/src/main/java/io/micronaut/rabbitmq/RabbitHeaders.java

```diff
@@ -0,0 +1,115 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.rabbitmq;
+
+import io.micronaut.core.convert.ArgumentConversionContext;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.messaging.MessageHeaders;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link MessageHeaders} implementation for Rabbit using a Map object.
+ *
+ * @author Raj Selvaraj
+ * @since 2.4.1
+ */
+public class RabbitHeaders implements MessageHeaders {
+
+    private final Map<String, Object> headers;
+
+    /**
+     * Constructs a new instance for the given map of headers.
+     *
+     * @param headers A Map of headers
+     */
+    public RabbitHeaders(Map<String, Object> headers) {
+        Objects.requireNonNull(headers, "Argument [headers] cannot be null");
+        this.headers = headers;
+    }
+
+    @Override
+    public List<String> getAll(CharSequence name) {
+        if (name != null) {
+            List<String> returnList = new ArrayList();
+            Object value = this.headers.get(name.toString());
+            if (value != null) {
+                returnList.add(value.toString());
+            }
+            return returnList;
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public String get(CharSequence name) {
+        if (this.headers != null && this.headers.get(name.toString()) != null) {
+            return this.headers.get(name.toString()).toString();
+        }
+        return null;
+    }
+
+    @Override
+    public Set<String> names() {
+        return this.headers.keySet();
+    }
+
+    @Override
+    public Collection<List<String>> values() {
+        return names().stream().map(name -> {
+            List<String> values = new ArrayList<>();
+            Object value = this.headers.get(name);
+            if (value != null) {
+                values.add(value.toString());
+            }
+            return values;
+        }).collect(Collectors.toList());
+    }
+
+    @Override
+    public <T> Optional<T> get(CharSequence name, ArgumentConversionContext<T> conversionContext) {
+        String v = get(name);
+        if (v != null) {
+            return ConversionService.SHARED.convert(v, conversionContext);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public RabbitHeaders add(CharSequence header, CharSequence value) {
+        if (header != null && value != null) {
+            this.headers.put(header.toString(), value.toString());
+        }
+        return this;
+    }
```

Comment:
You probably also want to provide a `add(CharSequence, Object)` method as well...

Reply:
Added add(CharSequence, Object) method